### PR TITLE
add equality operator for apf::Matrix

### PR DESCRIPTION
**Pull request by jacobmerson**
This functionality is needed for writing some unit tests. It uses
floating point arithmetic to find a close value. The isClose function
came from Knuth art of computer programming.

closes PR #144 since made the PR from the wrong base branch

### CODE REVIEW

File: apf/apfMatrix.h [hunk 1]
```diff
@@ -12,6 +12,20 @@
   \brief The APF linear algebra matrix interface */
 
 #include "apfVector.h"
+#include <limits> // epsilon value
+
+// this function doesn't use the "best" method for floating point comparison
+// FIXME update this with something like:
+// https://github.com/google/googletest/blob/dccc2d67547a5a3a97e4f211f39df931c6fbd5d5/googletest/include/gtest/internal/gtest-internal.h#L334-L347
+// the implemented expression comes from the art of computer programming by Knuth
```
Comment (cwsmith) [inline apf/apfMatrix.h]: Can you provide a more detailed reference?  Which volume chapter etc.. did it come from?  https://en.wikipedia.org/wiki/The_Art_of_Computer_Programming

File: apf/apfMatrix.h [hunk 2]
```diff
@@ -12,6 +12,20 @@
...
+// also see: https://stackoverflow.com/questions/17333/what-is-the-most-effective-way-for-float-and-double-comparison/3423299#3423299
```
Comment (cwsmith) [inline apf/apfMatrix.h]: We can/should remove the other links (google and stack overflow) and the FIXME comment.

**Comment by timetravellers** [comment]
Alright I removed the comments

File: apf/apfMatrix.h (hunk 1 above)
Comment (cwsmith) [reply apf/apfMatrix.h]: This isn't addressed.

**Pull request closed without merging**